src/devices/ExplorerHat/ExplorerHat.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Device.Gpio;
+using Iot.Device.ExplorerHat.Lighting;
+using Iot.Device.ExplorerHat.Motorization;
+
+namespace Iot.Device.ExplorerHat
+{
+    /// <summary>
+    /// Pimoroni Explorer HAT for Raspberry Pi
+    /// </summary>
+    public class ExplorerHat : IDisposable
+    {
+        private readonly bool _shouldDispose;
+        private GpioController _controller;
+
+        /// <summary>
+        /// Explorer HAT DCMotors collection
+        /// </summary>
+        public Motors Motors { get; private set; }
+
+        /// <summary>
+        /// Explorer HAT led array
+        /// </summary>
+        public Lights Lights { get; private set; }
+
+        /// <summary>
+        /// Initialize <see cref="ExplorerHat"/> instance
+        /// </summary>
+        public ExplorerHat(GpioController controller = null, bool shouldDispose = true)
+        {
+            _controller = controller;
+            _shouldDispose = shouldDispose;
+
+            if (_controller is null)
+            {
+                _controller = new GpioController(PinNumberingScheme.Logical);
+            }
+
+            if (_controller.NumberingScheme != PinNumberingScheme.Logical)
+            {
+                throw new ArgumentException("Invalid GpioController config: NumberingScheme value must be 'Logical'", nameof(controller));
+            }
+
+            Motors = new Motors(_controller);
+            Lights = new Lights(_controller);
+        }
+
+        #region IDisposable Support
+
+        private bool _disposedValue = false;
+
+        /// <summary>
+        /// Disposes the <see cref="ExplorerHat"/> instance
+        /// </summary>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
```

Comment:
you can set `_controller` to null after disposing and checking for that instead of extra variable